services/shap_explainer.py

```
@@ -216,11 +247,35 @@ def generate_force_plot(shap_values: shap.Explanation,
             values = shap_values.values[0]
         
+        # Prepare features vector to match shap_values length
+        features_vec = np.asarray(shap_values.data[0])
+        # If features_vec has extra dims (e.g., per-class rows), try to select the proper slice
+        if features_vec.ndim == 2:
+            # If second dim matches values length, try first row or class-row
+            if features_vec.shape[1] == np.asarray(values).ravel().shape[0]:
+                # If rows correspond to classes (n_classes x n_features) and pred_idx available, select that
+                try:
+                    pred_idx = np.argmax(values.sum()) if hasattr(values, 'shape') and values is not None else 0
+                except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)